scipy/optimize/optimize.py

```diff
@@ -2057,7 +2060,7 @@ def golden(func, args=(), brack=None, tol=_epsilon, full_output=0):
 
 
 def _minimize_scalar_golden(func, brack=None, args=(),
-                            xtol=_epsilon, **unknown_options):
+                            xtol=_epsilon, maxiter=500, **unknown_options):
```

Comment:
The default value probably should be bigger, so that you never run into it in normal operation and it only catches fully pathological cases. In the optimal case the interval at iteration `n` is `0.618034**n` smaller, so a default value of `maxiter=2000` is enough to underflow floating point --- so I would use that instead.